src/lsm/groove.zig

```diff
@@ -580,34 +581,89 @@ pub fn GrooveType(
             assert(groove.prefetch_ids.count() == 0);
         }
 
+        /// May be called on a Groove for which has_id is true.
+        ///
+        /// This function makes a best-effort attempt to ascertain the `absence` of the passed key
+        /// in the IdTree, without performing I/O. If the absence of the key cannot be ascertained
+        /// without I/O, it is enqueued for prefetching.
+        pub fn prefetch_enqueue_negative_lookup_optimized(groove: *Groove, key: PrimaryKey) void {
+            comptime assert(has_id);
+            if (!groove.ids.key_may_exist_in_tree(key)) return;
+            // TODO: After Fed's positive lookup changes, augment this to check (im)mutable table
+            // key range.
+            if (!groove.prefetch_from_memory_with_id(key)) {
+                const manifest_search_result = groove.ids.manifest_search(key);
+                switch (manifest_search_result) {
+                    .does_not_exist => {
+                        return;
+                    },
+                    .exists => {
+                        if (!groove.prefetch_from_memory_with_timestamp(manifest_search_result.exists.timestamp)) {
+                            groove.prefetch_ids.putAssumeCapacity(.{
+                                .timestamp = manifest_search_result.exists.timestamp,
+                            }, {});
+                        }
```

Comment:
Ah, I was looking for precisely this kind of usage of `switch` in our codebase, but wasn't able to find it. Just realized that I had missed this usage. :) 

This will make all of the `switch` statements significantly cleaner, thanks dj!